dashboard/app/assets/javascripts/client_state.js

```diff
@@ -136,5 +136,29 @@ dashboard.clientState.recordVideoSeen = function (videoId) {
   }
 };
 
+dashboard.clientState.hasSeenCallout = function(calloutId) {
+  var calloutsSeenJson = localStorage.getItem('calloutsSeen') || '{}';
+  try {
+    var calloutsSeen = JSON.parse(calloutsSeenJson);
+    return calloutsSeen[calloutId] == 1;
+  } catch (e) {
+    return false;
+  }
+};
+
+dashboard.clientState.recordCalloutSeen = function (calloutId) {
+  var calloutsSeenJson = localStorage.getItem('calloutsSeen') || '{}';
+
+  try {
```

Comment:
There's a lot of duplicated code in recordCalloutSeen and hasSeen callout for getting the calllout map.  It should be possible to get rid of it by factoring out a helper.